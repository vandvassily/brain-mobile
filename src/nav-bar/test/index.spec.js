import { mount } from '@vue/test-utils';
import NavBar from '..';

test('should render title slot and match snapshot', () => {
  const wrapper = mount(NavBar, {
    propsData: {
      title: 'Custom Title',
    },
  });

  expect(wrapper).toMatchSnapshot();
});

test('should emit click-left event when clicking left text', () => {
  const wrapper = mount(NavBar, {
    propsData: {
      leftText: 'left',
    },
  });

  wrapper.find('.nav-bar__left').trigger('click');
  expect(wrapper.emitted('click-left')).toBeTruthy();
});

test('should emit click-right event when clicking right text', () => {
  const wrapper = mount(NavBar, {
    propsData: {
      rightText: 'right',
    },
  });

  wrapper.find('.nav-bar__right').trigger('click');
  expect(wrapper.emitted('click-right')).toBeTruthy();
});
