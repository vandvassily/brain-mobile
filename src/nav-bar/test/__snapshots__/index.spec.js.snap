// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`left slot 1`] = `
<div class="nav-bar">
  <div class="nav-bar__content">
    <div class="nav-bar__left">
      <div>left</div>
    </div>
    <div class="nav-bar__title van-ellipsis"></div>
  </div>
</div>
`;

exports[`should render title slot and match snapshot 1`] = `
<div class="nav-bar">
  <div class="nav-bar__content">
    <div class="nav-bar__title van-ellipsis">Custom Title</div>
  </div>
</div>
`;
